RELEASE-NOTES.md

```diff
@@ -1,5 +1,27 @@
 # Prefect Release Notes
 
+## Release 2.10.17
+
+### Enhancements
+- Update PrefectClient to follow redirects by default — https://github.com/PrefectHQ/prefect/pull/9988
```

Comment:
```suggestion
- Update PrefectClient to follow redirects by default — https://github.com/PrefectHQ/prefect/pull/9988
- Always show checkboxes rather than animating them on hover — https://github.com/PrefectHQ/prefect-ui-library/pull/1490
```